test/conformance/conformance_flowgraph.h

```diff
@@ -17,74 +17,858 @@
+        assign_count = other.assign_count + 1;
+        ++other.assignes_count;
+        is_copy = true;
+        return *this;
+    }
+
+    CountingObject( CountingObject<O, I>&& other ):
+         copy_count(other.copy_count), copies_count(other.copies_count),
+         assign_count(other.assign_count), assignes_count(other.assignes_count),
+         move_count(other.move_count + 1), is_copy(other.is_copy) {}
+
+    CountingObject& operator=( CountingObject<O, I>&& other ) {
+        copy_count = other.copy_count;
+        copies_count = other.copies_count;
+        assign_count = other.assign_count;
+        assignes_count = other.assignes_count;
+        move_count = other.move_count + 1;
+        is_copy = other.is_copy;
+        return *this;
+    }
+
+    O operator()( I ) {
+        return O();
+    }
+
+    void operator()( I, multifunc_ports_t<I,O>& op ) {
+        std::get<0>(op).try_put(O());
+    }
+
+    void operator()( I , assync_ports_t<I, O>& g) {
+        g.try_put(O());
+    }
+
+    O operator()( oneapi::tbb::flow_control & fc ) {
+        static bool check = false;
+        if(check) {
+            check = false;
+            fc.stop();
+            return O();
+        }
+        check = true;
+        return O();
+    }
+};
+
+template <typename OutputType = int>
+struct passthru_body {
+    OutputType operator()( const oneapi::tbb::flow::continue_msg& ) {
+        return OutputType(0);
+    }
+
+    OutputType operator()( const OutputType& i ) {
+        return i;
+    }
+
+    OutputType operator()( oneapi::tbb::flow_control & fc ) {
+        static bool check = false;
+        if(check) {
+            check = false;
+            fc.stop();
+            return OutputType(0);
+        }
+        check = true;
+        return OutputType(0);
+    }
+
+    void operator()( OutputType argument, multifunc_ports_t<OutputType>& op ) {
+        std::get<0>(op).try_put(argument);
+    }
+
+    void operator()( OutputType argument, assync_ports_t<OutputType>& g ) {
+        g.try_put(argument);
+    }
+};
+
+template<typename Node, typename InputType, typename OutputType, typename ...Args>
+void test_body_exec(Args... node_args) {
+    oneapi::tbb::flow::graph g;
+    conformance::counting_functor<OutputType> counting_body;
+    counting_body.execute_count = 0;
+
+    Node node1(g, node_args..., counting_body);
 
+    constexpr std::size_t n = 10;
+    for(std::size_t i = 0; i < n; ++i) {
+#ifdef CONTINUE_NODE
+        CHECK_MESSAGE((node1.try_put(InputType()) == true),
+                "try_put of first node should return true");
+#else
+        CHECK_MESSAGE((node1.try_put(InputType(0)) == true),
+                "try_put of first node should return true");
+#endif
+    }
+    g.wait_for_all();
+
+    CHECK_MESSAGE((counting_body.execute_count == n), "Body of the first node needs to be executed N times");
+}
+
+template<typename Node, typename Body>
+void test_body_copying(Node& testing_node, Body& base_body) {
```

Comment:
There are instances of Counting Body with different template arguments, so in this case we need template `Body` argument in this function

Reply:
So maybe we may improve the code readability by using the `BodyType` template type instead of `Body`?
```suggestion
void test_body_copying(Node& testing_node, CountingBody<BodyType>& base_body) {
```